hybrid/SampleApps/NoteSync/src/com/salesforce/samples/notesync/ContentSoqlSyncTarget.java

```diff
@@ -0,0 +1,257 @@
+     * @param query
+     */
+    public ContentSoqlSyncTarget(String query) {
+        super(query);
+        this.queryType = QueryType.custom;
+    }
+
+    /**
+     * @return json representation of target
+     * @throws JSONException
+     */
+    public JSONObject asJSON() throws JSONException {
+        JSONObject target = super.asJSON();
+        target.put(ANDROID_IMPL, getClass().getName());
+        return target;
+    }
+
+    @Override
+    public JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws IOException, JSONException {
+        String queryToRun = maxTimeStamp > 0 ? SoqlSyncTarget.addFilterForReSync(getQuery(), maxTimeStamp) : getQuery();
+        syncManager.getRestClient().sendSync(RestRequest.getRequestForResources(syncManager.apiVersion)); // cheap call to refresh session
+        RestRequest request = buildQueryRequest(syncManager.getRestClient().getAuthToken(), queryToRun);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+        JSONArray records = parseSoapResponse(response);
+
+        return records;
+    }
+
+    @Override
+    public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSONException {
+        if (queryLocator == null) {
+            return null;
+        }
+        RestRequest request = buildQueryMoreRequest(syncManager.getRestClient().getAuthToken(), queryLocator);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+        JSONArray records = parseSoapResponse(response);
+
+        return records;
+    }
+
+    /**
+     * @param sessionId
+     * @param query
+     * @return request to run a soql query
+     * @throws UnsupportedEncodingException
+     */
+    private RestRequest buildQueryRequest(String sessionId, String query) throws UnsupportedEncodingException {
+        return buildSoapRequest(sessionId, String.format(QUERY_TEMPLATE, query));
+    }
+
+    /**
+     * @param sessionId
+     * @param locator
+     * @return request to do queryMore
+     * @throws UnsupportedEncodingException
+     */
+    private RestRequest buildQueryMoreRequest(String sessionId, String locator) throws UnsupportedEncodingException {
+        return buildSoapRequest(sessionId, String.format(QUERY_MORE_TEMPLATE, locator));
+    }
+
+    /**
+     *
+     * @param sessionId
+     * @param body
+     * @return request for soap call
+     * @throws UnsupportedEncodingException
+     */
+    private RestRequest buildSoapRequest(String sessionId, String body) throws UnsupportedEncodingException {
+        Map<String, String> customHeaders = new HashMap<String, String>();
+        customHeaders.put("SOAPAction", "\"\"");
+
+        StringEntity entity = new StringEntity(String.format(REQUEST_TEMPLATE, sessionId, body), HTTP.UTF_8);
+        entity.setContentType("text/xml");
+
+        return new RestRequest(RestRequest.RestMethod.POST, "/services/Soap/u/32.0", entity, customHeaders);
```

Comment:
Not a big deal, but could we read the ```32.0``` from ```ApiVersionStrings```. Makes it easier to update across the board.